# https://bugs.gentoo.org/949981
# https://www2.sqlite.org/src/info/ffd05de8a3b7cab1
From 09591b78ae28c6bb385ae96fef44edfb52454830 Mon Sep 17 00:00:00 2001
From: stephan <stephan@noemail.net>
Date: Mon, 24 Feb 2025 04:47:34 +0000
Subject: [PATCH] Update jimsh0.c from upstream autosetup to address
 [https://github.com/msteveb/autosetup/issues/75|autosetup ticket #75],
 triggered by this build in the Gentoo project.

FossilOrigin-Name: ffd05de8a3b7cab1b8e91b99da74b43ec5a7b91c78dcfd7cf338b66126e30a56
---
 autosetup/jimsh0.c | 20 ++++++++------------
 manifest           | 12 ++++++------
 manifest.uuid      |  2 +-
 3 files changed, 15 insertions(+), 19 deletions(-)

diff --git a/autosetup/jimsh0.c b/autosetup/jimsh0.c
index 0526b9a446..1a6453d0c8 100644
--- a/autosetup/jimsh0.c
+++ b/autosetup/jimsh0.c
@@ -1252,6 +1252,14 @@ int Jim_OpenForRead(const char *filename);
 
 #endif
 
+# ifndef MAXPATHLEN
+# ifdef PATH_MAX
+# define MAXPATHLEN PATH_MAX
+# else
+# define MAXPATHLEN JIM_PATH_LEN
+# endif
+# endif
+
 
 int Jim_FileStoreStatData(Jim_Interp *interp, Jim_Obj *varName, const jim_stat_t *sb);
 
@@ -2088,10 +2096,6 @@ enum wbuftype {
 #define UNIX_SOCKETS 0
 #endif
 
-#ifndef MAXPATHLEN
-#define MAXPATHLEN JIM_PATH_LEN
-#endif
-
 
 
 
@@ -4173,14 +4177,6 @@ int Jim_regexpInit(Jim_Interp *interp)
 #define S_ISDIR(m) (((m) & S_IFMT) == S_IFDIR)
 #endif
 
-# ifndef MAXPATHLEN
-# ifdef PATH_MAX
-# define MAXPATHLEN PATH_MAX
-# else
-# define MAXPATHLEN JIM_PATH_LEN
-# endif
-# endif
-
 #if defined(__MINGW32__) || defined(__MSYS__) || defined(_MSC_VER)
 #define ISWINDOWS 1
 
